crates/astria-sequencer/src/state_ext.rs

```diff
@@ -243,3 +243,298 @@ pub(crate) trait StateWriteExt: StateWrite {
 }
 
 impl<T: StateWrite> StateWriteExt for T {}
+
+#[cfg(test)]
+mod test {
+    use cnidarium::StateDelta;
+    use tendermint::Time;
+
+    use super::{
+        StateReadExt as _,
+        StateWriteExt as _,
+    };
+
+    #[tokio::test]
+    async fn chain_id() {
+        let storage = cnidarium::TempStorage::new()
+            .await
+            .expect("failed to create temp storage backing chain state");
+        let snapshot = storage.latest_snapshot();
+        let mut state = StateDelta::new(snapshot);
+
+        // doesn't exist at first
+        assert!(state.get_chain_id().await.is_err());
```

Comment:
I just using `expect_err("no chain ID should exist at first")` over `assert!` together with `Result::is_ok` or `Result::is_err`. In case the test is actually wrong the `is_*` methods just paper over what was actually inside (and that might be valuable).